Sources/Views/MessageLabel.swift

```diff
@@ -22,417 +22,491 @@
 
-        let origin = insetRect.origin
         let range = layoutManager.glyphRange(for: textContainer)
+        let origin = insetRect.origin
 
         layoutManager.drawBackground(forGlyphRange: range, at: origin)
         layoutManager.drawGlyphs(forGlyphRange: range, at: origin)
+        
     }
-
-    // MARK: - Public Methods
-
-    // MARK: UIGestureRecognizer Delegate
-
-    public func gestureRecognizer(_ gestureRecognizer: UIGestureRecognizer, shouldRecognizeSimultaneouslyWith otherGestureRecognizer: UIGestureRecognizer) -> Bool {
-        return true
+    
+    public func configure(block: () -> Void) {
+        isConfiguring = true
+        block()
+        isConfiguring = false
+        setNeedsDisplay()
     }
-
-    //swiftlint:disable cyclomatic_complexity
-    // Yeah we're disabling this because the whole file is a mess :D
-    public func gestureRecognizer(_ gestureRecognizer: UIGestureRecognizer, shouldReceive touch: UITouch) -> Bool {
-
-        let touchLocation = touch.location(in: self)
-
-        switch true {
-        case gestureRecognizer.view != self.superview && gestureRecognizer.view != self:
-            return true
-        case gestureRecognizer.view == self.superview:
-            guard let index = stringIndex(at: touchLocation) else { return true }
-            for (_, ranges) in rangesForDetectors {
-                for (nsRange, _) in ranges {
-                  guard let range = Range(nsRange) else { return true }
-                    if range.contains(index) { return false }
-                }
-            }
-            return true
-        case gestureRecognizer.view == self:
-            guard let index = stringIndex(at: touchLocation) else { return false }
-            for (_, ranges) in rangesForDetectors {
-                for (nsRange, _) in ranges {
-                    guard let range = Range(nsRange) else { return false }
-                    if range.contains(index) { return true }
-                }
+    
+    // MARK: - Setup
+    
+    private func setupTextContainer() {
+        textContainer.lineFragmentPadding = lineFragmentPadding
+        textContainer.maximumNumberOfLines = numberOfLines
+        textContainer.lineBreakMode = lineBreakMode
+        textContainer.size = frame.size
+    }
+    
+    private func setupLayoutManager() {
+        layoutManager.addTextContainer(textContainer)
+    }
+    
+    private func setupTextStorage() {
+        let attributedString = NSAttributedString(string: "")
+        _textSync = attributedString
+        textStorage.setAttributedString(attributedString)
+        textStorage.addLayoutManager(layoutManager)
+    }
+    
+    // MARK: - Manager TextStorage
+    
+    func updateTextStorage(text: NSAttributedString) {
+        
+        textStorage.setAttributedString(text)
+        
+        DispatchQueue.global(qos: .userInitiated).async {
+            
```

Comment:
Unless I use something like `NSOperation` to cancel these the entire effort is futile. Crashes as is and will likely lead to `TextStorage` corruption